src/openrct2-ui/windows/Ride.cpp

```diff
@@ -4778,7 +4808,24 @@ static void WindowRideColourPaint(rct_window* w, rct_drawpixelinfo* dpi)
                               (trackPreviewWidget.bottom + trackPreviewWidget.top) / 2 - 6 };
 
         ShopItem shopItem = rideEntry->shop_item[1] == ShopItem::None ? rideEntry->shop_item[0] : rideEntry->shop_item[1];
-        gfx_draw_sprite(dpi, ImageId(GetShopItemDescriptor(shopItem).Image, ride->track_colour[0].main), screenCoords);
+        static colour_t spriteColour = COLOUR_BLACK;
```

Comment:
If you have two shops windows open they will be on the same colour each refresh. I assume this isn't an issue but just bringing it up.

Reply:
Tweaked it a little more, should now be good.